include/seqan3/alignment/pairwise/detail/policy_affine_gap_recursion_banded.hpp

```diff
@@ -0,0 +1,104 @@
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::policy_affine_gap_recursion_banded.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <tuple>
+
+#include <seqan3/alignment/configuration/align_config_gap.hpp>
+#include <seqan3/alignment/matrix/detail/affine_cell_proxy.hpp>
+#include <seqan3/alignment/pairwise/detail/policy_affine_gap_recursion.hpp>
+#include <seqan3/alignment/pairwise/detail/type_traits.hpp>
+#include <seqan3/alignment/scoring/gap_scheme.hpp>
+#include <seqan3/core/type_traits/template_inspection.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Implements the alignment recursion function for the banded alignment algorithm using affine gap costs.
+ * \ingroup pairwise_alignment
+ * \copydetails seqan3::detail::policy_affine_gap_recursion
+ */
+template <typename alignment_configuration_t>
+class policy_affine_gap_recursion_banded : protected policy_affine_gap_recursion<alignment_configuration_t>
+{
+protected:
+    //!\brief The type of the base policy.
+    using base_policy_t = policy_affine_gap_recursion<alignment_configuration_t>;
+    // Import base types
+    using typename base_policy_t::traits_type;
+    using typename base_policy_t::score_type;
+    using typename base_policy_t::affine_cell_type;
+
+    //Import member types.
+    using base_policy_t::gap_extension_score;
+    using base_policy_t::gap_open_score;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    policy_affine_gap_recursion_banded() = default; //!< Defaulted.
+    policy_affine_gap_recursion_banded(policy_affine_gap_recursion_banded const &) = default; //!< Defaulted.
+    policy_affine_gap_recursion_banded(policy_affine_gap_recursion_banded &&) = default; //!< Defaulted.
+    policy_affine_gap_recursion_banded & operator=(policy_affine_gap_recursion_banded const &) = default; //!< Defaulted.
+    policy_affine_gap_recursion_banded & operator=(policy_affine_gap_recursion_banded &&) = default; //!< Defaulted.
+    ~policy_affine_gap_recursion_banded() = default; //!< Defaulted.
+
+    /*!\brief Construction and initialisation using the alignment configuration.
+     * \param[in] config The alignment configuration.
+     *
+     * \details
+     *
+     * Initialises the gap open score and gap extension score for this policy.
+     * If no gap cost model was provided by the user the default gap costs `-10` and `-1` are set for the gap open score
+     * and the gap extension score respectively.
+     */
+    explicit policy_affine_gap_recursion_banded(alignment_configuration_t const & config) : base_policy_t{config}
+    {}
+    //!\}
+
+    /*!\brief Initialises the first cell of a banded column that does not start in the first row of the matrix.
```

Comment:
it says here `... not start in the first row` but the function is called `initialise_band_first_row...`. Maybe a copy and paste error?

Reply:
Not a error but maybe I rename the function to make it more clear.